drivers/bmp280/BMP280.hpp

```diff
@@ -53,8 +57,13 @@ struct bmp280_sensor_calibration {
 	int16_t dig_P9;
 };
 
-
+#if defined(__DF_BBBLUE)
+#define BARO_DEVICE_PATH "/dev/i2c-2"
+extern pthread_mutex_t _mutex_shared_i2c_2_bus;
```

Comment:
Can you remove this mutex? Everything runs on a single thread.

Reply:
After removing the mutex, the following error occurred, so I added the mutex back.

px4 starting.

INFO  [dataman] Unknown restart, data manager file 'rootfs/fs/microsd/dataman' size is 11405132 bytes
INFO  [bbblue_adc] Initializing Robotics Cape library ...
344199 BMP280 sensor configuration succeeded
INFO  [df_bmp280_wrapper] BMP280 started
447748 MPU9250 sensor WHOAMI wrong: 0x0, should be: 0x71
ERROR [df_mpu9250_wrapper] MPU9250 start fail: -1
ERROR [df_mpu9250_wrapper] DfMpu9250Wrapper start failed
Command 'df_mpu9250_wrapper' failed, returned -1
INFO  [bbblue_adc] BBBlueADC started

Also note that "bbblue_adc start" is placed after bmp280 & mpu9250 wrapper, the log message of bbblue_adc came first.

BTW, in pull request https://github.com/PX4/Firmware/pull/9635, Lorenz Meier mentioned moving from DriverFramework to CDev soon, so these 2 drivers will be replaced eventually.

Reply:
Might be that some startup code is called from the wrong thread then. Ok, let's keep it in under the assumption that we want to move away from DriverFramework.